### PR TITLE
Added JUnit tests for core/exception/model package

### PR DESCRIPTION
**Pull request by sonja-ct**
This PR contains JUnit tests for classes in exception/model package in
rest-api/core module:
 - IllegalArgumentExceptionInfo class
 - IllegalNullArgumentExceptionInfo class
 - InternalUserOnlyExceptionInfo class
 - MfaRequiredExceptionInfo class
 - SelfManagedOnlyExceptionInfo class
 - SubjectUnauthorizedExceptionInfo class
 - ThrowableInfo class

Signed-off-by: Sonja <sonja.matic@endava.com>

**Related Issue**
/

**Description of the solution adopted**
/

**Screenshots**
/

**Any side note on the changes made**
/

### CODE REVIEW

**Review by LeoNerdoG — APPROVED** [review]
hey @Coduz,
I have checked this PR and it looks OK. This is part of "rest api unit tests" that we have been talking about (achieving code coverage for rest-api module through unit tests). 
After all these PRs are merged and we have framework for Rest Api testing in place, we will add high-level/integration rest api tests, that will check the actual fucntionality. 
Please take a look and merge when you can.

**Review by LeoNerdoG — APPROVED** [review]

File: rest-api/core/src/test/java/org/eclipse/kapua/app/api/core/exception/model/MfaRequiredExceptionInfoTest.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                assertEquals("Expected and actual values should be the same.", expectedKapuaErrorCode[j], mfaRequiredExceptionInfo.getKapuaErrorCode());
+                assertEquals("Expected and actual values should be the same.", expectedStatusCodes[i], mfaRequiredExceptionInfo.getHttpErrorCode());
+                assertEquals("Expected and actual values should be the same.", "Error: ", mfaRequiredExceptionInfo.getMessage());
+            }
+        }
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void mfaRequiredExceptionInfoNullStatusExceptionTest() {
+        for (KapuaAuthenticationErrorCodes code : errorCodes) {
```
Comment (Coduz) [inline rest-api/core/src/test/java/org/eclipse/kapua/app/api/core/exception/model/MfaRequiredExceptionInfoTest.java]: Why using a for-loop with `@Test(expected = NullPointerException.class)`?

File: rest-api/core/src/test/java/org/eclipse/kapua/app/api/core/exception/model/MfaRequiredExceptionInfoTest.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            kapuaException = new KapuaAuthenticationException(code);
+            new MfaRequiredExceptionInfo(null, kapuaException);
+        }
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void mfaRequiredExceptionInfoStatusNullExceptionTest() {
+        for (Response.Status status : statusList) {
```
Comment (Coduz) [inline rest-api/core/src/test/java/org/eclipse/kapua/app/api/core/exception/model/MfaRequiredExceptionInfoTest.java]: Why using a for-loop with `@Test(expected = NullPointerException.class)`?

**Review by Coduz — CHANGES_REQUESTED** [review]

**Comment by sonja-ct** [comment]
Hey @Coduz I made requested changes, so you can merge this when you have time. Thanks

**Pull request merged**